database-migration/src/main/java/uk/ac/ebi/eva/mongo/migration/MongoMigrationMain.java

```diff
@@ -0,0 +1,144 @@
+
+        System.out.println(String.format("mongodb://%s", mongoUri));
+
+        Mongobee runner = new Mongobee(String.format("mongodb://%s", mongoUri));
+        runner.setDbName(dbName);  // host must be set if not set in URI
+        runner.setChangeLogsScanPackage("uk.ac.ebi.eva.mongo.migration"); // package to scan for changesets
+        runner.setEnabled(true);         // optional: default is true
+        runner.execute();
+    }
+
+    @ChangeSet(order = "001", id = "migrateStatistics", author = "Diego")
+    public void migrateStatistics(MongoTemplate mongoTemplate) {
+        logger.info("1) migrate statistics");
+
+        final DBCollection statisticsCollection = mongoTemplate.getCollection("statistics");
+        final DBCollection variantsCollection = mongoTemplate.getCollection(variantsCollectionName);
+
+        BulkWriteOperation bulkInsertMaf = variantsCollection.initializeUnorderedBulkOperation();
+        BulkWriteOperation bulkInsertStats = statisticsCollection.initializeUnorderedBulkOperation();
+
+        int counter=0;
+        int missingMafFlag = Integer.MIN_VALUE;
+
+        DBCursor variantCursor = variantsCollection.find();
+        while (variantCursor.hasNext()) {
+
+            DBObject variantObj = variantCursor.next();
+
+            DBObject statsObj = (DBObject) variantObj.get("st");
+
+            if(statsObj!=null){
+                for(String s : statsObj.keySet()){
+                    BasicDBObject iStatsObj = (BasicDBObject) statsObj.get(s);
+
+                    iStatsObj.put("vid", variantObj.get("_id"));
+                    iStatsObj.put("chr", variantObj.get("chr"));
+                    iStatsObj.put("start", variantObj.get("start"));
+                    iStatsObj.put("ref", variantObj.get("ref"));
+                    iStatsObj.put("alt", variantObj.get("alt"));
+
+                    bulkInsertStats.insert(iStatsObj);
+
+                    //
+                    double maf = iStatsObj.getDouble("maf",  missingMafFlag);
+                    if(maf != missingMafFlag){
+                        DBObject update = new BasicDBObject("$addToSet", new BasicDBObject("maf", maf));
+                        DBObject find = new BasicDBObject("_id", variantObj.get("_id"));
+                        bulkInsertMaf.find(find).updateOne(update);
+                    }
+
+                    counter++;
+                    if(counter % 1000 == 0){
+                        bulkInsertStats.execute();
+                        bulkInsertStats = statisticsCollection.initializeUnorderedBulkOperation();
+                        bulkInsertMaf.execute();
+                        bulkInsertMaf = variantsCollection.initializeUnorderedBulkOperation();
+                    }
+                }
+            }
+
+        }
+
+        if(counter % 1000 != 0){
+            bulkInsertStats.execute();
+            bulkInsertMaf.execute();
+        }
+
```

Comment:
Ideas on how to ensure the next changeset is not run unless the count of read and written statistics documents match? There doesn't seem to be a standardized way to do it.